flight/hal/argus_v2.py

```diff
@@ -0,0 +1,803 @@
+            #     ArgusV2Components.TORQUE_ZP_POWER_MONITOR_I2C_ADDRESS,
+            #     ArgusV2Components.TORQUE_ZP_POWER_MONITOR_I2C,
+            # ],
+            # "TORQUE_ZM": [
+            #     ArgusV2Components.TORQUE_ZM_POWER_MONITOR_I2C_ADDRESS,
+            #     ArgusV2Components.TORQUE_ZM_POWER_MONITOR_I2C,
+            # ],
+            # "SOLAR_XP": [
+            #     ArgusV2Components.SOLAR_CHARGING_XP_POWER_MONITOR_I2C_ADDRESS,
+            #     ArgusV2Components.SOLAR_CHARGING_XP_POWER_MONITOR_I2C,
+            # ],
+            # "SOLAR_XM": [
+            #     ArgusV2Components.SOLAR_CHARGING_XM_POWER_MONITOR_I2C_ADDRESS,
+            #     ArgusV2Components.SOLAR_CHARGING_XM_POWER_MONITOR_I2C,
+            # ],
+            # "SOLAR_YP": [
+            #     ArgusV2Components.SOLAR_CHARGING_YP_POWER_MONITOR_I2C_ADDRESS,
+            #     ArgusV2Components.SOLAR_CHARGING_YP_POWER_MONITOR_I2C,
+            # ],
+            # "SOLAR_YM": [
+            #     ArgusV2Components.SOLAR_CHARGING_YM_POWER_MONITOR_I2C_ADDRESS,
+            #     ArgusV2Components.SOLAR_CHARGING_YM_POWER_MONITOR_I2C,
+            # ],
+            # "SOLAR_ZP": [
+            #     ArgusV2Components.SOLAR_CHARGING_ZP_POWER_MONITOR_I2C_ADDRESS,
+            #     ArgusV2Components.SOLAR_CHARGING_ZP_POWER_MONITOR_I2C,
+            # ],
+        }
+
+        from hal.drivers.adm1176 import ADM1176
+
+        error_codes = []
+
+        for location, busAndAddress in locations.items():
+            try:
+                address = busAndAddress[0]
+                bus = busAndAddress[1]
+                power_monitor = ADM1176(bus, address)
+
+                if self.__middleware_enabled:
+                    power_monitor = Middleware(power_monitor)
+
+                self.__power_monitors[location] = power_monitor
+                self.__device_list.append(power_monitor)
+                error_codes.append(Errors.NOERROR)  # Append success code if no error
+            except Exception as e:
+                self.__power_monitors[location] = None
+                if self.__debug:
+                    print(f"Failed to initialize {location} power driver: {e}")
+                    raise e
+                return Errors.ADM1176_NOT_INITIALIZED
+
+        return error_codes
+
+    def __imu_boot(self) -> list[int]:
+        """imu_boot: Boot sequence for the IMU
+
+        :return: Error code if the IMU failed to initialize
+        """
+        try:
+            # from hal.drivers.bno08x_i2c import BNO08X_I2C
+            from hal.drivers.bno085 import BNO085, BNO_REPORT_ACCELEROMETER, BNO_REPORT_GYROSCOPE, BNO_REPORT_MAGNETOMETER
+
+            imu = BNO085(ArgusV2Components.IMU_I2C, ArgusV2Components.IMU_I2C_ADDRESS)
+            imu.enable_feature(BNO_REPORT_ACCELEROMETER)
+            imu.enable_feature(BNO_REPORT_GYROSCOPE)
+            imu.enable_feature(BNO_REPORT_MAGNETOMETER)
+            if self.__middleware_enabled:

```

Comment:
See @NeilKhera note on slack. The raw measurements of the bno must be enabled here @perrintong325 @Akshat412